app/Http/Controllers/AnnouncementController.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Announcement;
+use Illuminate\Http\Request;
+
+class AnnouncementController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('auth');
+    }
+
+    public function index()
+    {
+        return view('announcement');
+    }
+    public function store(Request $request)
+    {
+       $data = $request->validate([
+            'date' => 'required',
+            'description' => 'required',
+            'location' => 'required',
+        ]);
+
+        Announcement::create(['user_id' => auth()->id(), 'description' => $data['description'], 'date' => $data['date'], 'where' => $data['location']]);
```

Comment:
you can break this in multiple line, to make it easy to read.

eg.
Announcement::create([
    //columns here
])